paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/samples/ui/customersheet/CustomerSheetExampleViewModel.kt

```diff
@@ -40,27 +41,33 @@ class CustomerSheetExampleViewModel(
         customerEphemeralKeyProvider = {
             fetchCustomerEphemeralKey().fold(
                 success = {
-                    Result.success(
+                    CustomerAdapterResult.Success(
```

Comment:
I like the idea of factories (same as what kotlin.Result is doing) for instantiating this. Not blocking, but I'd strongly encourage it.